### PR TITLE
SAK-41065 -  Switching itext to openpdf

### PR DESCRIPTION
**Pull request by jonespm**
So I was looking at this and I don't see why we need this overridden [HTMLWorker](https://github.com/sakaiproject/sakai/blob/master/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/pdf/itext/HTMLWorker.java) anymore.

The three things I noticed as a difference
* There is an unused method `getImageStream` which refers to Sakai stuff. 
* The constructor checks if pdf.default.font is set and sets it. Since the variable set (cprops) is private we can't set this on the class that already extends this.
` String fontName = ServerConfigurationService.getString("pdf.default.font");`

This code was added in [SAM-2276](https://sakaiproject.atlassian.net/browse/SAK-37252) to support latin characters. However I tested this both with and without this library and it didn't make a difference with latin character. Perhaps it's already supported fine upstream.

* There was a change on [SAM-2660 ](https://sakaiproject.atlassian.net/browse/SAK-36424) about external images. But these also don't seem to be an issue as I can add images and the PDF is fine.

[HTMLWorker](https://github.com/LibrePDF/OpenPDF/blob/master/openpdf/src/main/java/com/lowagie/text/html/simpleparser/HTMLWorker.java) has changed quite a bit since we forked it. I think ideally we'd want to pick up those changes. I created to branches, one that uses HTMLWorker and [one that keeps it with modifications.](https://github.com/jonespm/sakai/commit/66b88c60d5ffdf3c1f1105336e8ba8751d21ca87) I think this is the better one. 

One thing I did notice is that [Chinese/Japanese characters are still](https://sakaiproject.atlassian.net/browse/SAK-21908) not supported. I don't see this as a regression. Maybe have to [look more into this later. ](https://github.com/LibrePDF/OpenPDF/wiki/Multi-byte-character-language-support-with-TTF-fonts)

### CODE REVIEW

File: samigo/samigo-app/pom.xml [hunk 1]
```diff
@@ -97,23 +97,9 @@
...
-                    <artifactId>bcmail-jdk14</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>bouncycastle</groupId>
-                    <artifactId>bcprov-jdk14</artifactId>
-                </exclusion>
-            </exclusions>
+            <groupId>com.github.librepdf</groupId>
+            <artifactId>openpdf</artifactId>
+            <version>1.3.26</version>
```
[the remark this answers is not in the document]
Comment (josecebe) [reply samigo/samigo-app/pom.xml:102]: > Could we put the version in a Maven variable inside master/pom.xml? I'd like to stick with the same version.
> 
> Thanks and good job so far!

Hi @jonespm , I have added the variable to master/pom.xml, it's named "sakai.openpdf.version".

File: jsf/jsf-spreadsheet/pom.xml [hunk 2]
```diff
@@ -45,23 +45,9 @@
...
-                    <artifactId>bcmail-jdk14</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>bouncycastle</groupId>
-                    <artifactId>bcprov-jdk14</artifactId>
-                </exclusion>
-            </exclusions>
+            <groupId>com.github.librepdf</groupId>
+            <artifactId>openpdf</artifactId>
+            <version>1.3.26</version>
```
[the remark this answers is not in the document]
Comment (jonespm) [reply jsf/jsf-spreadsheet/pom.xml]: Thanks, updated!

File: samigo/samigo-app/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonespm) [reply samigo/samigo-app/pom.xml]: Thanks, updated!

**Comment by jonespm** [comment]
Looks like just one more itext in kernel-impl is being deployed from 
`fr.opensagres.xdocreport.converter.odt.odfdom:jar`

A new version of this released in Jan 2022 but still had this dependency. I feel like we can get rid of this but I'll create a new issue for it.

**Review by mpellicer — APPROVED** [review]

**Pull request merged**